src/Sulu/Bundle/ContentBundle/Controller/ContentController.php

```diff
@@ -0,0 +1,347 @@
+
+/**
+ * Provides api for content querying.
+ */
+class ContentController extends RestController implements ClassResourceInterface
+{
+    const WEBSPACE_NODE_SINGLE = 'single';
+    const WEBSPACE_NODES_ALL = 'all';
+
+    private static $relationName = 'content';
+
+    use RequestParametersTrait;
+
+    /**
+     * @var ContentRepositoryInterface
+     */
+    private $contentRepository;
+
+    /**
+     * @var ViewHandlerInterface
+     */
+    private $viewHandler;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var WebspaceManagerInterface
+     */
+    private $webspaceManager;
+
+    /**
+     * @var SessionManagerInterface
+     */
+    private $sessionManager;
+
+    public function __construct(
+        ContentRepositoryInterface $contentRepository,
+        ViewHandlerInterface $viewHandler,
+        RouterInterface $router,
+        WebspaceManagerInterface $webspaceManager,
+        SessionManagerInterface $sessionManager,
+        TokenStorageInterface $tokenStorage = null
+    ) {
+        $this->contentRepository = $contentRepository;
+        $this->viewHandler = $viewHandler;
+        $this->router = $router;
+        $this->webspaceManager = $webspaceManager;
+        $this->sessionManager = $sessionManager;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Returns content array by parent or webspace root.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws MissingParameterException
+     * @throws ParameterDataTypeException
+     */
+    public function cgetAction(Request $request)
+    {
+        $parent = $request->get('parent');
+        $properties = array_filter(explode(',', $request->get('mapping', '')));
+        $excludeGhosts = $this->getBooleanRequestParameter($request, 'exclude-ghosts', false, false);
+        $excludeShadows = $this->getBooleanRequestParameter($request, 'exclude-shadows', false, false);
+        $webspaceNodes = $this->getRequestParameter($request, 'webspace-nodes');
+        $locale = $this->getRequestParameter($request, 'locale', true);
+        $webspaceKey = $this->getRequestParameter($request, 'webspace', false);
+
+        if (!$webspaceKey && !$webspaceNodes) {
+            throw new MissingParameterException(
+                get_class($this), sprintf('"%s" or "%s" or "%s"', 'webspace', 'webspace-nodes', 'webspace-nodes')
+            );
+        }
+
+        if (!in_array($webspaceNodes, [self::WEBSPACE_NODE_SINGLE, self::WEBSPACE_NODES_ALL, null])) {
+            throw new ParameterDataTypeException(get_class($this), 'webspace-nodes');
+        }
+
+        $user = $this->tokenStorage->getToken()->getUser();
+
+        $mapping = MappingBuilder::create()
+            ->disableHydrateGhost(!$excludeGhosts)
+            ->disableHydrateShadow(!$excludeShadows)
```

Comment:
Double negations like this are very hard to read... Can't you reverse this logic, to be more "positive"?